lib/logging/logging.dart

```diff
@@ -331,11 +343,24 @@ class LogDetailsUI extends CoreElement {
   LogData get data => _data;
 
   set data(LogData value) {
+    // TODO(devoncarew): Reset the vertical scroll value if any.
+
     _data = value;
 
     if (_data != null) {
-      // TODO(dantup): Can we format the JSON better?
-      message.text = _data.message;
+      final String str = _data.message;
+
+      if (str.startsWith('{') && str.endsWith('}')) {
+        try {
+          // If the string decodes properly, than format the json.
+          final dynamic result = jsonDecode(str);
+          message.text = jsonEncoder.convert(result);
```

Comment:
Will the logs potentially ever include strings that are JSON that we don't want to re-format? (for ex. if users logs can appear in this view, do we want their JSON to be reformatted? it might be confusing, though also it could be useful?)